### PR TITLE
Update upper bound for galaxy packages to < 23.2

### PR DESCRIPTION
**Pull request by bernt-matthias**
... waiting for the  23.1 release :)

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -3,8 +3,8 @@ bioblend>=1.0.0
 click!=8.0.2
 cwltool>=1.0.20191225192155
 ephemeris>=0.10.3
-galaxy-tool-util>=23.0,<23.1
-galaxy-util[template]>=23.0,<23.1
+galaxy-tool-util>=23.0,<24.0
+galaxy-util[template]>=23.0,<24.0
```
Comment (nsoranzo) [inline requirements.txt:7]: I don't think we have officially ruled out a 23.2 release.

```suggestion
galaxy-tool-util>=23.0,<23.2
galaxy-util[template]>=23.0,<23.2
```

**Comment by bernt-matthias** [comment]
Test is failing reproducibly. But not sure if related.

**Comment by bgruening** [comment]
The error I can see is:

```
E           PackagesNotFoundError: The following packages are not available from current channels:
E           
E             - fleeqtk
E           
E           Current channels:
E           
E             - https://conda.anaconda.org/conda-forge/linux-64
E             - https://conda.anaconda.org/conda-forge/noarch
E             - https://conda.anaconda.org/bioconda/linux-64
E             - https://conda.anaconda.org/bioconda/noarch
E             - https://repo.anaconda.com/pkgs/main/linux-64
E             - https://repo.anaconda.com/pkgs/main/noarch
E             - https://repo.anaconda.com/pkgs/r/linux-64
E             - https://repo.anaconda.com/pkgs/r/noarch
```

Which is true. This packages does not exist, its a fake package that is build locally during a test.

I think `--conda_use_local` is not working in this PR.

**Comment by bernt-matthias** [comment]
Bumping conda helps with the failing test. Let's see if there are side effects https://github.com/galaxyproject/galaxy/pull/17013

**Comment by nsoranzo** [comment]
I am not sure how https://github.com/galaxyproject/galaxy/pull/17013 can help, it bumps the *minimum* conda version but in tests it's most likely that the latest conda version is installed either way.

**Comment by bernt-matthias** [comment]
All I know is that in the failing test on my local computer not the latest was installed, but:

```
           conda version : 23.3.1
    conda-build version : 3.27.0
```

When I bumped the minimum requirement the tests were successful. No idea how to test this here, without bumping the Galaxy packages first. Can we somehow patch tool-util during the test?

**Comment by nsoranzo** [comment]
> All I know is that in the failing test on my local computer not the latest was installed, but:
> 
> ```
>            conda version : 23.3.1
>     conda-build version : 3.27.0
> ```

Is that what you have in `~/miniconda3` ?

> When I bumped the minimum requirement the tests were successful. No idea how to test this here, without bumping the Galaxy packages first. Can we somehow patch tool-util during the test?

Did you update the conda package manually? Changing `CONDA_PACKAGE_SPECS` only affects new conda installations.

**Comment by nsoranzo** [comment]
What's weird when looking at the CI logs is that `test_local_conda_dependencies_version` failed at the `planemo test` stage, when the `__fleeqtk@1.3` conda environment should have been created already by `planemo conda_install` right before.

**Comment by bernt-matthias** [comment]
> What's weird when looking at the CI logs is.. 

Yes. They had exit code 0, despite the failure.

**Comment by mvdbeek** [comment]
Can you check in the github workflow if there is any interference from the conda or .condarc already installed on the VM ?

**Comment by bernt-matthias** [comment]
So, now we see the error from the build step. Seems that this installs 23.3.1 .. my hypothesis is as follows: 

- the [mambaforge installer](https://github.com/galaxyproject/galaxy/blob/2319bc31af19726c4f8596621f1e01950da6f0c8/lib/galaxy/tool_util/deps/conda_util.py#L67) comes with preinstalled conda 23.3.1 (just verified)
- when installing conda use the two-step approach
  1. try to fulfill requirements such that installed packages don't need to be updated
  2. if this is not possible fulfill requirements and allow updating installed packages

so with our current conda pins, we are stuck at 23.3.1 (maybe also because we install in the base env?) and not the newest packages are installed. 

Does this sound reasonable?

Note that also `planemo conda_search` seems not to use the exit code of the `exec_command` function. If desired I could try to fix it here.

**Comment by nsoranzo** [comment]
Thanks @bernt-matthias , the latest commit to check the exit code really helped clarifying the issue, and your explanation makes sense. I didn't expect the latest Mambaforge to install conda 23.3.1, but any way it looks like that https://github.com/galaxyproject/galaxy/pull/17013 is the correct solution here.

> Note that also `planemo conda_search` seems not to use the exit code of the `exec_command` function. If desired I could try to fix it here.

Yes, that would be good, thanks!

**Comment by nsoranzo** [comment]
planemo/commands/cmd_conda_install.py also has the same issue.

**Comment by bernt-matthias** [comment]
Does anyone have an idea what's going on with the 3.7 tests?

**Comment by mvdbeek** [comment]
dev doesn't work on 3.7, that's why I rebased your PR

**Review by mvdbeek — APPROVED** [review]
Very cool, thanks @bernt-matthias

**Pull request merged**